txmongo/utils/__init__.py

```diff
@@ -57,7 +58,7 @@ def on_fail(failure):
     return _timeout
 
 
-def check_deadline(_deadline):
+def check_deadline(_deadline: Optional[float]):
```

Comment:
Should we switch away from `Optional[float]` and just use `float | None`; and typing in general? Or keep that change for another PR?

Reply:
Would love to, but `T | S` syntax is only for 3.10+ and `list[T]` instead of `List[T]` is 3.9+